### PR TITLE
CTCP-5476: Implemented dynamic content for OtherThingsToReport page depending on answers to revised unloading procedure (new auth) page and seals replaced by customs authority page

### PR DESCRIPTION
**Pull request by kevincoatsworth**

### CODE REVIEW

**Comment by kevincoatsworth** [comment]
Old auth - Report page:
Old auth - Report page

Old auth - Report page with required error:
Old auth - Report page with required error

Old auth - Report page with invalid error:
Old auth - Report page with invalid error

Old auth - Report page with length error:
Old auth - Report page with length error

Old auth - CYA label for report page:
Old auth - CYA label for report page

New auth - Report page:
New auth - Report page

New auth - Report page with required error:
New auth - Report page with required error

New auth - Report page with invalid error:
New auth - Report page with invalid error

New auth - Report page with length error:
New auth - Report page with length error

New auth - Report page link to NewAuthYesNo page:
New auth - Report page link to NewAuthYesNo page

New auth - CYA label for report page:
New auth - CYA label for report page

**Comment by Reece-Carruthers** [comment]
It looks like there's an extra space just have procedure showing "revised unloading procedure ."
image

**Comment by kevincoatsworth** [comment]
@Reece-Carruthers I believe that's just because the full stop is sitting after the link element:
```
<a id="link" class="govuk-link" href=${routes.NewAuthYesNoController.onSubmit(arrivalId, mode)}>
        ${messages("otherThingsToReport.newAuth.link")}
    </a>.
```

**Comment by Reece-Carruthers** [comment]
@kevincoatsworth yeah, I wonder if a .trim() would get rid of the white space at the end, might be worth a try, if not happy to go with the above since we have it elsewhere

<a id="link" class="govuk-link" href=${routes.NewAuthYesNoController.onSubmit(arrivalId, mode)}>
    ${messages("otherThingsToReport.newAuth.link")}.trim()
</a>.

**Comment by kevincoatsworth** [comment]
@Reece-Carruthers tried this: 
${messages("otherThingsToReport.newAuth.link").trim()}
and it just looks the same. I think it's because there's not actually like a whitespace character there. It's more like padding from the element

**Review by Reece-Carruthers — APPROVED** [review]

**Comment by kevincoatsworth** [comment]
After this ticket was developed it was revised to cater for 3 scenarios and I've been advised to close this PR.

**Comment by kevincoatsworth** [comment]
Old auth - report page:
Old auth - report page

Old auth - report page with error required:
Old auth - report page with error required

Old auth - report page with invalid error:
Old auth - report page with invalid error

Old auth - report page with length required:
Old auth - Report page with length error

New auth + no replaced seals - Report page:
New auth + no replaced seals - Report page

New auth + no replaced seals - Report page with required error:
New auth + no replaced seals - Report page with required error

New auth + no replaced seals - Report page with invalid error:
New auth + no replaced seals - Report page with invalid error

New auth + no replaced seals - Report page with length error:
New auth + no replaced seals - Report page with length error

New auth + no replaced seals - Report page link:
New auth + no replaced seals - Report page link

New auth with replaced seals - Report page:
New auth with replaced seals - Report page

New auth with replaced seals - Report page with required error:
New auth with replaced seals - Report page with required error

New auth with replaced seals with invalid error:
New auth with replaced seals with invalid error - Report page

New auth with replaced seals - Report page with length error:
New auth with replaced seals - Report page with length error

New auth with replaced seals - Report page link:
New auth with replaced seals - Report page link

File: test/views/OtherThingsToReportViewSpec.scala [hunk 1]
```diff
@@ -18,32 +18,102 @@ package views
...
+  private val hint = "Each seal can be up to 20 characters long and include both letters and numbers."
+
+  "when newAuth is false" - {
+    val viewModel = new OtherThingsToReportViewModelProvider().apply(arrivalId, mode, newAuth = false, sealsReplaced = None)
+    val doc = parseView(
+      injector.instanceOf[OtherThingsToReportView].apply(form, mrn, arrivalId, otherThingsToReportLength, NormalMode, viewModel)(fakeRequest, messages)
+    )
+
+    behave like pageWithoutHint(doc, hint)
+  }
```
Comment (jackheslop96) [inline test/views/OtherThingsToReportViewSpec.scala:72]: Don't use the `ViewModelProvider` here. We don't want this test to be dependent on the logic in that apply method. All we want to do here is define a OtherThingsToReportViewModel with hint = None and assert that no hint gets rendered

File: test/views/OtherThingsToReportViewSpec.scala [hunk 2]
```diff
@@ -18,32 +18,102 @@ package views
...
+      doc = doc,
+      id = "link",
+      expectedText = "select no to using the revised unloading procedure",
+      expectedHref = s"/manage-transit-movements/unloading/$arrivalId/revised-unloading-procedure"
+    )
+
+    behave like pageWithPartialContent(doc, "p", ". You will then need to unload the goods and report any discrepancies.")
+
+    behave like pageWithHint(doc, hint)
+  }
```
Comment (jackheslop96) [inline test/views/OtherThingsToReportViewSpec.scala:95]: As above

File: test/views/OtherThingsToReportViewSpec.scala [hunk 3]
```diff
@@ -18,32 +18,102 @@ package views
...
```
Comment (jackheslop96) [inline test/views/OtherThingsToReportViewSpec.scala:118]: As above

**Pull request merged**